api4/file.go

```diff
@@ -298,16 +294,39 @@ func getPublicFile(c *Context, w http.ResponseWriter, r *http.Request) {
 	}
 }
 
-func writeFileResponse(filename string, contentType string, bytes []byte, toDownload bool, w http.ResponseWriter, r *http.Request) *model.AppError {
+func writeFileResponse(filename string, contentType string, bytes []byte, forceDownload bool, w http.ResponseWriter, r *http.Request) *model.AppError {
 	w.Header().Set("Cache-Control", "max-age=2592000, private")
 	w.Header().Set("Content-Length", strconv.Itoa(len(bytes)))
+	w.Header().Set("X-Content-Type-Options", "nosniff")
+
+	if contentType == "application/javascript" || contentType == "application/ecmascript" ||
+		contentType == "text/javaScript" || contentType == "text/ecmascript" ||
+		contentType == "application/x-javascript" || contentType == "text/html" {
+		contentType = "application/octet-stream"
+	}
 
 	if contentType != "" {
 		w.Header().Set("Content-Type", contentType)
 	} else {
 		w.Header().Del("Content-Type") // Content-Type will be set automatically by the http writer
```

Comment:
yeah, so actually i am gonna request this change since this does circumvent the blacklisting logic. you can confirm it by uploading an html file without an extension